admin/tool/policy/classes/api.php

```diff
@@ -121,7 +123,11 @@ public static function get_policy_version($policyid, $versionid) {
                        v.id AS versionid, v.usermodified, v.timecreated, v.timemodified, v.revision, v.content, v.contentformat
                   FROM {tool_policy} d
                   JOIN {tool_policy_versions} v ON v.policyid = d.id
-                 WHERE d.id = :policyid AND v.id = :versionid";
+                 WHERE v.id = :versionid";
+
+        if ($policyid) {
+            $sql .= ' AND d.id = :policyid';
+        }
```

Comment:
Hmm. I was thinking about this and I actually thought it would be better to demand both policyid and versionid on the API level, to force and implicitly check the data consistency. Maybe I was wrong. In either case, if the policyid is not provided, we should not set it in the `$params` too, so that the number of placeholders matches the number of provided values.

Reply:
You are right, this is a reasonable change in the API. I just made a follow up commit fc4deca2389eb737a2820c16e46c5e5738091211 to fix the params and added unit tests for the both cases.

Thanks!